packages/js/src/api/http-client.ts

```diff
@@ -18,20 +20,31 @@ export class HttpClient {
       apiVersion = DEFAULT_API_VERSION,
       apiUrl = DEFAULT_BACKEND_URL,
       userAgent = DEFAULT_USER_AGENT,
+      headers = {},
     } = options || {};
     this.apiVersion = apiVersion;
     this.apiUrl = `${apiUrl}/${this.apiVersion}`;
     this.headers = {
       'Novu-API-Version': NOVU_API_VERSION,
       'Content-Type': 'application/json',
       'User-Agent': userAgent,
+      ...headers,
     };
   }
 
   setAuthorizationToken(token: string) {
     this.headers.Authorization = `Bearer ${token}`;
   }
 
+  setKeylessHeader() {
+    const identifier = window.localStorage.getItem('novu_keyless_application_identifier');
+    if (!identifier) {
+      return;
+    }
+
+    this.headers['Novu-Application-Identifier'] = identifier;
+  }
```

Comment:
Been thinking out load, what will happen if the user started in keyles, and then onboarded? Will this key be removed and we won't use it anymore? Meaning a real appId takes priority over the keyless?

Reply:
good point! 
i pushed a fix.